numpy/lib/function_base.py

```diff
@@ -1320,6 +1320,477 @@ def gradient(f, *varargs, **kwargs):
     else:
         return outvals
 
+def divergence(v,*varargs,**kwargs):
+    """
+    Return the divergence of an N-dimensional vector field of N 
+    components, each of dimension N.
+
+    The divergence is computed using second order accurate central
+    differences in the interior and either first differences or second
+    order accurate one-sides (forward or backwards) differences at the
+    boundaries. The returned gradient hence has the same shape as the
+    input array.
+    
+    Parameters
+    ----------  
+    v : list of numpy arrays
+        Each of these array is the N-dimensional projection of the vector
+        field on the corresponding axis. 
+    
+    varargs : scalar or list of scalar, optional
+        N scalars specifying the sample distances for each dimension,
+        i.e. `dx`, `dy`, `dz`, ... Default distance: 1.
+        single scalar specifies sample distance for all dimensions.
+        if `axis` is given, the number of varargs must equal the number of axes.
+    edge_order : {1, 2}, optional
```

Comment:
This seems to not actually be implemented/make a difference currently. Which also points to an unfortunate huge chunk of work, but it can wait for a while, just needs patience. And that is, we need a lot of tests after hashing out what we want.